fix(runtime/testing): false positive for timers when an error is thrown

caspervonb opened the pull request:
Fixes https://github.com/denoland/deno/issues/8954

bartlomieju reviewed (changes requested):
@caspervonb I just run provided test case on `main` branch and it results in exactly same output from `deno test`. I think this solution is preferable but we need another test case that actually demonstrates what's being fixed

bartlomieju reviewed (approved):
LGTM, thanks Casper, provided test case fails on `main`

caspervonb commented:
Oops I cleaned up that test a bit too much, until it wasn't testing anything.
Fixed up test fails all cases on 1.7.5
 
![Screenshot 2021-02-22 at 12 10 19 AM](https://user-images.githubusercontent.com/157787/108630923-5dabdb80-74a2-11eb-9857-1742ea5d616c.png)
